test/Microsoft.IdentityModel.JsonWebTokens.Tests/JwtTokenUtilitiesTests.cs

```diff
@@ -208,6 +215,281 @@ public void ResolveTokenSigningKey()
         }
 
+        #region DecryptJwtToken Tests
+
+        [Fact]
+        public void GetCryptoProviderFactory_WhenSetOnTokenValidationParameters_ReturnsFromValidationParameters()
+        {
+            // Arrange
+            // SecurityKey can never have 'null' CryptoProviderFactory.
+            var securityKey = symmetricSecurityKey;
+            var validationParameters = new TokenValidationParameters()
+            {
+                CryptoProviderFactory = new CryptoProviderFactory(),
+            };
+
+            // Act
+            var result = JwtTokenUtilities.s_getCryptoProviderFactory(validationParameters, securityKey);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Same(validationParameters.CryptoProviderFactory, result);
+        }
+
+        [Fact]
+        public void GetCryptoProviderFactory_WhenSetToNullOnTokenValidationParameters_ReturnsFromSecurityKey()
+        {
+            // Arrange
+            // SecurityKey can never have 'null' CryptoProviderFactory.
+            var securityKey = symmetricSecurityKey;
+            var validationParameters = new TokenValidationParameters
+            {
+                CryptoProviderFactory = null,
+            };
+
+            // Act
+            var result = JwtTokenUtilities.s_getCryptoProviderFactory(validationParameters, securityKey);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(securityKey.CryptoProviderFactory, result);
+        }
+
+        [Fact]
+        public void DecryptJwtToken_WhenValidationParametersIsNull_ShouldThrow()
+        {
+            // Arrange
+            TokenValidationParameters validationParameters = null;
+
+            // Act & Assert
+            var exception = Assert.Throws<ArgumentNullException>(() =>
+                JwtTokenUtilities.DecryptJwtToken(
+                    securityToken: null,
+                    validationParameters: validationParameters,
+                    decryptionParameters: null));
+
+            Assert.Equal("validationParameters", exception.ParamName);
+        }
+
+        [Fact]
+        public void DecryptJwtToken_WhenJwtTokenDecryptionParametersIsNull_ShouldThrow()
+        {
+            // Arrange
+            var validationParameters = new TokenValidationParameters();
+            JwtTokenDecryptionParameters decryptionParameters = null;
+
+            // Act & Assert
+            var exception = Assert.Throws<ArgumentNullException>(() =>
+                JwtTokenUtilities.DecryptJwtToken(
+                    securityToken: null,
+                    validationParameters: validationParameters,
+                    decryptionParameters: decryptionParameters));
+
+            Assert.Equal("decryptionParameters", exception.ParamName);
+        }
+
+        [Fact(Skip = "Skip due to potential false positives it may cause running tests in parallel as it overrides JwtTokenUtilities.s_getCryptoProviderFactory.")]
```

Comment:
Skip is not needed.

`By default, each test class is a unique test collection. Tests within the same test class will not run in parallel against each other. Let's examine a very simple test assembly, one with a single test class:`

https://xunit.net/docs/running-tests-in-parallel#:~:text=By%20default%2C%20each%20test%20class%20is%20a%20unique%20test%20collection.%20Tests%20within%20the%20same%20test%20class%20will%20not%20run%20in%20parallel%20against%20each%20other.%20Let%27s%20examine%20a%20very%20simple%20test%20assembly%2C%20one%20with%20a%20single%20test%20class%3A

Reply:
Thanks for the pointer. I wasn't aware of. Removed skip.